reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -410,7 +425,35 @@ def generate_xy_plane(n_pix):
         return xy_plane
 
     @staticmethod
-    def generate_slices(map_3d_f, xy_plane, rots):
+    def generate_xyz_voxels(n_pix):
+        """Generate (x,y,z) cube.
+
+        x, y, z axis values range [-n // 2, ..., n // 2 - 1]
+
+        Parameters
+        ----------
+        n_pix : int
+            Number of pixels along one edge of the cube.
+
+        Returns
+        -------
+        xyz : arr
+            Array describing xyz cube in space.
+            Shape (3, n_pix**3)
+        """
+        axis_pts = np.arange(-n_pix // 2, n_pix // 2)
+        grid = np.meshgrid(axis_pts, axis_pts, axis_pts)
+
+        xyz = np.zeros((3, n_pix**3))
+
+        for di in range(3):
+            xyz[di] = grid[di].flatten()
+        xyz[[0, 1]] = xyz[[1, 0]]
+
+        return xyz
+
+    @staticmethod
+    def generate_slices(map_3d_f, xy_plane, rots, z_offset=0.05):
```

Comment:
why `z_offset=0.05`? It might more sense to do one full voxel away

Reply:
One full voxel introduces a whole heap of slice coordinates that we don't want to overwrite - this solves the problem without making the slice any less 2D in effect.